### DISCUSSION TITLE
Allow alert rules in the same folder to have the same name

### DISCUSSION THREAD

**Discussion by M0rdecay** [grafana/grafana#45307] [Alerting]
Right now, the alert name is taken directly from the rule name. Allow to overwrite the name of the alert with the appropriate label.

In our case, we have a few alerts with the same name but different query/label details. In version 7, it was allowed to have several alerts with the same name, but in version 8, this is not allowed if the rules are located in the same directory.

**Comment by M0rdecay** [comment]
It looks like it is enough to move this line a little higher
https://github.com/grafana/grafana/blob/main/pkg/services/alerting/notifiers/alertmanager.go#L145

Upd. Ooops, my apologize, we're talking about `ngalert` here

If I'm not mistaken (correct me), `alertname` label is set here
https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/state/cache.go#L89

The question is - it's okay to save `alertname` label in this place, or would it cause some kind of error?
```go
func (c *cache) getOrCreate(ctx context.Context, alertRule *ngModels.AlertRule, result eval.Result) *State {
	c.mtxStates.Lock()
	defer c.mtxStates.Unlock()

	// clone the labels so we don't change eval.Result
	labels := result.Instance.Copy()
+++	alertname, found := labels[prometheusModel.AlertNameLabel]
	attachRuleLabels(labels, alertRule)
	ruleLabels, annotations := c.expandRuleLabelsAndAnnotations(ctx, alertRule, labels, result)
	// or?
+++	alertname, found := ruleLabels[prometheusModel.AlertNameLabel]

	// if duplicate labels exist, alertRule label will take precedence
	lbs := mergeLabels(ruleLabels, result.Instance)
	attachRuleLabels(lbs, alertRule)

+++	// keep original alert name if the corresponding label was specified
+++	if found {
+++		lbs[prometheusModel.AlertNameLabel] = alertname
+++	}

	il := ngModels.InstanceLabels(lbs)
....
}
```

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by hackery** [comment]
It previously *was* issue #45308 - maybe reopen that?
